pkg/trait/pull_secret.go

```diff
@@ -122,6 +122,10 @@ func (t *pullSecretTrait) newImagePullerRoleBinding(e *Environment) *rbacv1.Role
 		serviceAccount = "default"
 	}
 	return &rbacv1.RoleBinding{
```

Comment:
I wonder if there would be a logical owner reference for that resource, so that it gets garbage collected when no needed anymore. Maybe the IntegrationPlatform could be the owner?

Also I think there is an issue when two Integrations in the same namespace use different service accounts, as the RoleBinding will be updated with the last one. I wonder if a single RoleBinding maintaining all the subjects for the global platform could work, but that'd probably be more involved to implement.

Reply:
Right, previously the resource was created directly by the trait, but a [recent change](https://github.com/apache/camel-k/commit/17cf52fc72a518fe44506ae97514a32ed444ce1c#diff-ffe17df62455fe57d1838af003526f25a73d77e84e950fdb275ea162e7699d96L114-L131) made it pass from the trait cycle, so it got updated in the owner trait.

I think it's ok to attach them to the integrationplatform, +1.

The problem with multiple sas is real.. To simplify things, we can use a different naming convention, attaching the name of the service account to the rolebinding's name (rather than managing multiple subjects), so that we can have multiple ones without issues.

Reply:
Ah looking at the offending commit, I wonder whether it might been better to revert back to having the trait managing the resource directly instead of adding it the environment resources. It's not a typical resource, and it seems safer to manage it directly to avoid any new side effect in the future.